### DISCUSSION TITLE
Only connect and Visually attach pieces to together if correct and get completeness progress

### DISCUSSION THREAD

**Discussion by Devdre1909** [flbulgarelli/headbreaker#90] [Q&A] [answered]
Hi! Thank you for building Headbreaker, it really helps! 

I noticed that any pieces can connect with any other pieces, which creates a problem for me. I need to track progress to get a percentage of completeness, and I would prefer that only pieces that are actually correct are connected together.

**Comment by flbulgarelli** [comment, marked as answer]
Hi! Thanks for your comments. 

I suggest checking this question https://github.com/flbulgarelli/headbreaker/discussions/61, which describes how to prevent wrong connections.

**Reply by Devdre1909** [reply]
Okay, awesome.

Checking out it.

What about tracking progress?

**Reply by flbulgarelli** [reply]
You could register a callback that will be called each time a piece is connected or disconnected using `onConnect` and `onDisconnect`. You'll need to manually track the number of connected pieces, which could be used of an indicator of progress.
